### PR TITLE
[Nure] Replace nukeeper by nuget calls, move to .net 5

### PR DESCRIPTION
**Pull request by comagnan**
This is the big one. With previous behavior, we were launching nukeeper to bump dependencies, which comes with the following limitations:
- Errors while bumping libraries caused nukeeper to stop its update process, but temporary errors are unavoidable due to transitive dependencies
- It's impossible to target a specific library with nukeeper, so we could not dodge the issue manually
- Nukeeper exits with an error code of 0 in any situation, making it difficult to know the process was a failure in the first place

This is replaced a rudimentary loop over the dependencies obtained from a dotnet list until everything is bumped. By replacing nukeeper, we're also able to use .net 5 instead of .net core 2, which saves us from some irony.

See NuGetWrapper for the new behavior.

Also added here:
- Custom users/emails in the nure config
- Cleaner branch names

### CODE REVIEW

File: Nure.UnitTests/Update/RuntimeParametersTests.cs [hunk 1]
```diff
@@ -14,23 +15,42 @@ public void ValidRuntimeParametersAreAccepted()
...
+            };
 
             Assert.False(missingPassword.Validate());
             Assert.False(missingPath.Validate());
             Assert.False(missingUser.Validate());
         }
 
-        private static RuntimeParameters GivenValidRuntimeParameters()
+        [Fact]
+        public void MaliciousRuntimeParametersAreDenied()
```
Comment (ebrazier) [inline Nure.UnitTests/Update/RuntimeParametersTests.cs:37]: good catch

File: Nure/Configuration/NureOptions.cs [hunk 2]
```diff
@@ -4,42 +4,36 @@ namespace Nure.Configuration
...
-        public NureOptions()
-        {
-            AllowPrereleaseDependencies = DEFAULT_ALLOW_PRELEASE_DEPENDENCIES;
-            CustomNugetSources = new List<string>();
-            PackagesToIgnore = new List<string>();
-            CommitMessage = DEFAULT_COMMIT_MESSAGE_PREFIX;
-            PullRequestTitle = DEFAULT_PULL_REQUEST_TITLE;
-        }
+        private const string DEFAULT_COMMIT_USER = "Jenkins";
+        private const string DEFAULT_COMMIT_EMAIL = "email@email.com";
```
Comment (ebrazier) [inline Nure/Configuration/NureOptions.cs:10]: Does this matter at all or is it a placeholder?

File: Nure/PullRequest/Bitbucket/BitbucketClient.cs [hunk 3]
```diff
@@ -44,22 +44,13 @@ public BitbucketUser GetCurrentUser()
...
-                .GetJsonAsync<BitbucketUser>()
-                .Result;
+            return new Url(m_BaseUrl).AppendPathSegments(API_VERSION, "user").WithBasicAuth(m_Username, m_Password).WithClient(m_FlurlClient).GetJsonAsync<BitbucketUser>().Result;
         }
 
         public List<BitbucketUser> GetDefaultReviewers()
         {
             s_Logger.Info($"Getting default reviewers of repository {m_RepositoryId} from Bitbucket.");
-            BitbucketPagedResult<BitbucketUser> reviewers = new Url(m_BaseUrl)
-                .AppendPathSegments(API_VERSION, "repositories", m_WorkspaceId, m_RepositoryId, "default-reviewers")
```
Comment (ebrazier) [inline Nure/PullRequest/Bitbucket/BitbucketClient.cs:58]: Very nit picky, but is there a way  to parametrize the  URL building?
Some repeat and some dont  e.g. `.WithBasicAuth(m_Username, m_Password)
                .WithClient(m_FlurlClient)`

File: Nure/Update/NuGetWrapper.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+                if (regexMatch.Success) {
+                    isReadingPackages = false;
+
+                    currentProject = m_MonitoredProjects.First(project => project.Contains(regexMatch.Value, StringComparison.InvariantCultureIgnoreCase));
+                    packagesByProject.Add(currentProject, new List<NuGetPackage>());
+                } else if (isReadingPackages) {
+                    List<string> packageComponents = currentLine.Split(" ", StringSplitOptions.RemoveEmptyEntries).ToList();
+
+                    if (packageComponents.Count == 5) {
```
Comment (ebrazier) [inline Nure/Update/NuGetWrapper.cs:70]: Wuuuut
do you have a line example? I take it the split will always return 5?

File: Nure/Update/NuGetWrapper.cs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+using System.Text.RegularExpressions;
+using NLog;
+using Nure.Configuration;
+
+namespace Nure.Update
+{
+    public class NuGetWrapper
+    {
+        private const string NUGET_PACKAGE_HEADER = "Top-level Package";
+        private const string DOTNET_PROJECT_SEPARATOR = "----------";
```
Comment (ebrazier) [inline Nure/Update/NuGetWrapper.cs:15]: :notsureif:

File: Nure/Configuration/NureOptions.cs (hunk 2 above)
Comment (comagnan) [reply Nure/Configuration/NureOptions.cs:10]: Placeholder to support existing lure configs 👍

File: Nure/Update/NuGetWrapper.cs (hunk 4 above)
Comment (comagnan) [reply Nure/Update/NuGetWrapper.cs:70]: image

File: Nure/Update/NuGetWrapper.cs (hunk 5 above)
Comment (comagnan) [reply Nure/Update/NuGetWrapper.cs:15]: image

😅

**Review by ebrazier — APPROVED** [review]

**Comment by ebrazier** [comment]
:shipit:

**Pull request merged**